test/parser/test_required_fields.py

```diff
@@ -0,0 +1,338 @@
+from typing import Any, Dict
+import pytest
+import test.parser.schema_with_required_fields as schemas
+from harambe.parser.parser import (
+    PydanticSchemaParser,
+    SchemaValidationError,
+    RequiredFieldsError,
+)
+
+
+@pytest.mark.parametrize(
+    "schema, data",
+    [
+        (
+            schemas.document_schema,
+            {
+                "title": None,  # required
+                "document_url": None,  # required
+            },
+        ),
+        (
+            schemas.documents_schema,
+            {
+                "documents": [
+                    {
+                        "title": "",  # required
+                        "document_url": "",  # required
+                    }
+                ],
+            },
+        ),
+        (
+            schemas.contact_schema,
+            {
+                "name": {
+                    "first_name": None,  # required
+                    "last_name": "test",
+                },
+                "address": {
+                    "street": "test",
+                    "city": None,  # required
+                    "zip": "test",
+                },
+                "phone_numbers": [
+                    {
+                        "type": "test",
+                        "number": "test",
+                    }
+                ],
+            },
+        ),
+        (
+            schemas.list_of_strings_schema,
+            {
+                "tags": [],  # required
```

Comment:
Hmm... I'm not sure this should be an error case...

I guess it should be fine. Array fields must always have [] by default. I guess we can make required here mean it must have at least 1 item